File: dataset_reduzido/codenet/Java/Code/s493101281.java
import java.util.*;
import java.io.*;

public class Main {
    public static final long MOD = 1000000007;

    public static void main(String[] args) {
        FastScanner sc = new FastScanner();
        PrintWriter out = new PrintWriter(System.out);
        Integer i,j,k,n=0,m=0,h=0,w=0,ans=0;
        long a=0,b=0,c=0,d=0,x=0,y=0,z=0;
        List<Integer> l = new ArrayList<>();
        List<String> s = new ArrayList<>();
        List<Pair> p = new ArrayList<>();

        n=sc.nextInt();

        for(i=0;i<n;i++){
            l.add(sc.nextInt());
        }

        Collections.sort(l);

        print(l.get(n/2)-l.get(n/2-1));

        out.flush();
    }

    private static void print(Object o) {
        System.out.println(o.toString());
    }

    private static long lcm(long m, long n) {
        return m * n / gcd(m, n);
    }

    private static long gcd(long m, long n) {
        if(m < n) return gcd(n, m);
        if(n == 0) return m;
        return gcd(n, m % n);
    }

    private static class Pair{
        private int key;
        private int value;

        public Pair(int key, int value){
            this.key=key;
            this.value=value;
        }

        public int getValue(){
            return this.value;
        }

        public int getKey(){
            return this.key;
        }
    }
}

class FastScanner {
    private final InputStream in = System.in;
    private final byte[] buffer = new byte[1024];
    private int ptr = 0;
    private int buflen = 0;
    private boolean hasNextByte() {
        if (ptr < buflen) {
            return true;
        }else{
            ptr = 0;
            try {
                buflen = in.read(buffer);
            } catch (IOException e) {
                e.printStackTrace();
            }
            if (buflen <= 0) {
                return false;
            }
        }
        return true;
    }
    private int readByte() { if (hasNextByte()) return buffer[ptr++]; else return -1;}
    private static boolean isPrintableChar(int c) { return 33 <= c && c <= 126;}
    public boolean hasNext() { while(hasNextByte() && !isPrintableChar(buffer[ptr])) ptr++; return hasNextByte();}
    public String next() {
        if (!hasNext()) throw new NoSuchElementException();
        StringBuilder sb = new StringBuilder();
        int b = readByte();
        while(isPrintableChar(b)) {
            sb.appendCodePoint(b);
            b = readByte();
        }
        return sb.toString();
    }
    public long nextLong() {
        if (!hasNext()) throw new NoSuchElementException();
        long n = 0;
        boolean minus = false;
        int b = readByte();
        if (b == '-') {
            minus = true;
            b = readByte();
        }
        if (b < '0' || '9' < b) {
            throw new NumberFormatException();
        }
        while(true){
            if ('0' <= b && b <= '9') {
                n *= 10;
                n += b - '0';
            }else if(b == -1 || !isPrintableChar(b)){
                return minus ? -n : n;
            }else{
                throw new NumberFormatException();
            }
            b = readByte();
        }
    }
    public int nextInt() {
        long nl = nextLong();
        if (nl < Integer.MIN_VALUE || nl > Integer.MAX_VALUE) throw new NumberFormatException();
        return (int) nl;
    }
    public double nextDouble() { return Double.parseDouble(next());}
}

File: output/LLaMa_repaired_1/codenet/Java/Python/s493101281.py
MOD = 1000000007

class FastScanner:
    def __init__(self):
        self.buffer = ''
        self.ptr = 0

    def next(self):
        while not self.buffer:
            self.buffer = sys.stdin.readline().strip()
        token, self.buffer, self.ptr = self.buffer.split()[0], self.buffer[self.ptr:], self.ptr + 1
        return token

    def next_int(self):
        return int(self.next())

    def next_long(self):
        return int(self.next())

    def next_double(self):
        return float(self.next())

class Pair:
    def __init__(self, key, value):
        self.key = key
        self.value = value

    def get_value(self):
        return self.value

    def get_key(self):
        return self.key

gcd_cache = {}

def gcd(a, b):
    if a < b:
        a, b = b, a
    if a in gcd_cache and b in gcd_cache[a]:
        return gcd_cache[a][b]
    elif b == 0:
        return a
    else:
        gcd_cache[a] = gcd_cache.get(a, {})
        gcd_cache[a][b] = gcd(b, a % b)
        return gcd_cache[a][b]

def lcm(a, b):
    return a * b // gcd(a, b)

def print_o(o):
    print(str(o))

def main():
    fs = FastScanner()
    n = fs.next_int()
    l = sorted([fs.next_int() for _ in range(n)])
    print(o := l[n // 2] - l[n // 2 - 1])

if __name__ == "__main__":
    main()